include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -84,8 +84,29 @@ A_{3, 2}  A_{3, 3}  A_{3, 0}  A_{3, 1} ...   [phase 1] /
         auto mfmaEnc = dotOpEnc.getParent().dyn_cast<MfmaEncodingAttr>();
 
         if (mfmaEnc) {
-          // Swizzling is currently disabled for MFMA
-          return $_get(context, 1, 1, 1, order);
+          int kDimNum = dotOpEnc.getOpIdx() == 0 ? 1 : 0;
+          bool isKDimInner = (order[0] == kDimNum);
+          if (isKDimInner) {
+            const int numBanks = 32;
+            const int bankBitWidth = 32;
+
+            // number of inner dimension rows per one pattern repeat
+            int outerDimGranularity = mfmaEnc.getNonKDim();
+            int typeBitWidth = eltTy.getIntOrFloatBitWidth();
+            int innerDimLength = shape[order[0]];
+            int elemsPerOneBanksRow = (numBanks * bankBitWidth) / typeBitWidth;
+
+            int perPhase = std::max(1, elemsPerOneBanksRow / innerDimLength);
+            int maxPhase = outerDimGranularity / perPhase;
```

Comment:
We can set `maxPhase` * `perPhase` = 16 instead of 32 since only 16 (the width of SIMD unit) workitems access LDS at the same cycle. 
This further reduces `maxPhase`, which can simply the XOR function a little.

Reply:
This is interesting!
I'll take a look

Reply:
Unfortunately, looks like there are 32 simultaneous accesses.

I've experimented with it: https://gist.github.com/binarman/717494a48bb215f9786897a5af0c6302
and got bank conflicts if 32 threads tried to access same 16 banks in one load operation

Reply:
If there are 32 simultaneous accesses, there should be 1 conflict per access, instead of 0.02, which seems to be a very small given the size of the example kernel.
The information about "only 16 workitems access LDS per cycle due to SIMD16" was discussed here: https://confluence.amd.com/pages/viewpage.action?spaceKey=MLSE&title=LDS+bank+conflicts

Reply:
Thank you for the link!
I'll read it carefully